src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -70,7 +68,11 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
 
         // Note: Source files are fingerprinted as part of the serialized logical plan currently.
         val sourceDataProperties =
-          Hdfs.Properties(Content("", Seq(Content.Directory("", sourceFiles, NoOpFingerprint()))))
+          Hdfs.Properties(
+            Content(
+              "",
+              Seq(
+                Content.Directory("", sourceFiles(df), sourceRelations(df), NoOpFingerprint()))))
```

Comment:
seems need to retrieve & keep the root path and use the path for spark.read.load at RefreshAction
let me update this + clean up the previous ver.

Reply:
Can you clarify this?

If `df` was created from `spark.read.parquet("/tmp/1/1.parquet", "/tmp/2/2.parquet")`, what will be the root path used for refresh? (Note that there could be `/tmp/1/3.parquet`).

cc: @apoorvedave1

Reply:
I'm going to collect the root paths by using `location.rootPaths.map(_.toString),` and add "rootPaths: Seq[String]" to RelationMetadataEntry.

Reply:
Ah OK. You meant `rootPaths` from `FileIndex`.